### PR TITLE
Correct optimize created interventions

### PR DESCRIPTION
**Pull request by Tom-Szendrey**
# Description
- Optimize returns results as list this needed to be updated to reflect that
- Add more typing for readability

# Testing:
Run optimize and check that the result is a properly formatted intervention
You can run with one or many interventions
You can select any type of (start time, param value, both)

### CODE REVIEW

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts [hunk 1]
```diff
@@ -253,6 +270,7 @@ export async function createInterventionPolicyFromOptimize(
 	modelConfigId: string,
 	optimizeRunId: string
 ): Promise<InterventionPolicy | null> {
+	console.log('Test');
```
Comment (dvince2) [inline packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts:273]: `Test`

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts [hunk 2]
```diff
@@ -163,15 +163,26 @@ export const OptimizeCiemssOperation: Operation = {
...
+	}
+
 	const allInterventions: Intervention[] = [];
 	// Get the interventionPolicyGroups from the simulation object.
 	// This will prevent any inconsistencies being passed via knobs or state when matching with result file.
 	const simulation = await getSimulation(optimizeRunId);
 
 	const simulationStaticInterventions: any[] = simulation?.executionPayload.fixed_static_parameter_interventions ?? [];
-	const optimizeInterventions = simulation?.executionPayload?.optimize_interventions;
+	const optimizeInterventions: OptimizeInterventionsCamelCase[] = simulation?.executionPayload?.optimize_interventions;
```
Comment (dvince2) [inline packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts:183]: does this not come through server to fix the camel case issue?

**Review by dvince2 — APPROVED** [review]

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts (hunk 1 above)
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts:273]: IM IN DRAFT! loool
Thanks i was just removing

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts (hunk 2 above)
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts:183]: As discussed this `executionPayload` is untyped 
I am doubtful we will end up typing it out as the interfaces have been updated a lot but we would likely want to :

abstract ExecutionPayload<T> with one T per simulation type
I believe the complete list is:
```
simulate
calibrate
optimize
ensemble simulate
ensemble calibrate
funman validation 
```

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts [hunk 3]
```diff
@@ -163,15 +163,26 @@ export const OptimizeCiemssOperation: Operation = {
...
+	// This is a camel case version of types/Types OptimizeInterventions as we are reading from simulation execution payload
+	interface OptimizeInterventionsCamelCase {
+		intervention_type: string;
+		param_names: string[];
+		param_values?: number[];
+		start_time?: number[];
+		objective_function_option: string;
+		initial_guess?: number[];
+		relative_importance: number;
+	}
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts:175]: Your comment says camel-case, ditto for the interface name,  but this is snake-case so I am not sure what you are trying to say here.

**Review by mwdchang — APPROVED** [review]

**Pull request merged**